pulp_python/app/pypi/views.py

```diff
@@ -235,14 +252,55 @@ class SimpleView(PackageUploadMixin, ViewSet):
         ],
     }
 
+    def perform_content_negotiation(self, request, force=False):
+        """
+        Uses standard content negotiation, defaulting to HTML if no acceptable renderer is found.
+        """
+        try:
+            return super().perform_content_negotiation(request, force)
+        except NotAcceptable:
+            return TemplateHTMLRenderer(), "text/html"
+
+    def get_renderers(self):
+        """
+        Uses custom renderers for PyPI Simple API endpoints, defaulting to standard ones.
+        """
+        if self.action in ["list", "retrieve"]:
+            # Ordered by priority if multiple content types are present
+            return [
+                TemplateHTMLRenderer(),
+                PyPISimpleHTMLRenderer(),
+                PyPISimpleJSONRenderer(),
+            ]
+        else:
+            return [JSONRenderer(), BrowsableAPIRenderer()]
+
+    def handle_exception(self, exc):
+        # todo: fixes test_pull_through_filter?
+        if isinstance(exc, (Http404, NotFound)):
+            # Force JSON response since JSONRenderer is not in get_renderers()
+            return JsonResponse({"detail": str(exc)}, status=404)
+        return super().handle_exception(exc)
```

Comment:
There's no reason the 404 has to be a JSON response. You could change the test to just check for the error string instead of expecting the result to be json.